Sources/TuistKit/Services/InitService.swift

```diff
@@ -92,10 +101,29 @@ class InitService {
 
         let directories = try templatesDirectoryLocator.templateDirectories(at: path)
         if let templateName = templateName {
+            var template: Template?
+            if templateName.isURL {
+                do {
+                    let temporaryDirectory = try fileHandler.temporaryDirectory()
+                        .appending(component: "Template")
+                    try fileHandler.createFolder(temporaryDirectory)
+                    try gitHandler.clone(url: templateName, to: temporaryDirectory)
```

Comment:
We should use `checkout` or a shallow clone as we don't need to clone the whole repo

Reply:
Could also download the zip directly (i.e. https://github.com/jrescabias/cliTemplate/archive/refs/heads/main.zip) rather than using `git` commands.

Reply:
I like this approach - however, we'd have to know in advance/determine which branch is the default if I'm not mistaken?

Reply:
You can get that information from the Github API (i.e. https://api.github.com/repos/jrescabias/cliTemplate) using the `default_branch` key.

Reply:
would be interesting let the user choose specific branch with other optional argument? if not use `default_branch`

Reply:
You could add another argument, but I think a better approach would be to support urls which link to a specific branch.

Reply:
sorry for delay, I've been resting for some days. Still thinking about this. I prefer use `git` commands over download a zip. By default the checkout will be at the `main` branch, as I suggested we can create an optional argument to checkout other branch instead `main`. Something like `tuist init -t <repo_URL> -b develop`. Let me know what do you guys think. Hugs!